### PR TITLE
MF-1154 - Move UUID provider to pkg

### LINKED ISSUE #1154: Move identity providers to project root
Mutualize Things, Authn, and Users identity providers

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

Resolves #1154

### CODE REVIEW

**Review by mteodor — DISMISSED** [review]
LGTM

**Review by blokovi — DISMISSED** [review]

File: authn/service_test.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ const (
 
 func newService() authn.Service {
 	repo := mocks.NewKeyRepository()
-	idp := mocks.NewIdentityProvider()
+	idp := mocks.NewUUIDProvider()
```
Comment (drasko) [inline authn/service_test.go]: I do not like the idea to call this var `idp` any more

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by mteodor — DISMISSED** [review]

**Comment by nmarcetic** [comment]
@mainflux/maintainers Why not moving this to `/pkg` it's a great candidate as a shared package.

**Comment by manuio** [comment]
@nmarcetic I'm not against. I also wanted to know where do you think that we should put uuid mock. In `pkg/uuid/mocks` directory or in a specific `pkg/mocks` directory?

**Comment by nmarcetic** [comment]
> @nmarcetic I'm not against. I also wanted to know where do you think that we should put uuid mock. In `pkg/uuid/mocks` directory or in a specific `pkg/mocks` directory?

If those mocks are not sharable (e.g not used by there packages, but only by `/pkg/uuid`) then I would keep them under `/pkg/uuid`

**Review by drasko — APPROVED** [review]
LGTM

**Review by nmarcetic — APPROVED** [review]

**Pull request merged**